src/main/java/seedu/address/model/person/ModuleCode.java

```diff
@@ -1,12 +1,25 @@
 package seedu.address.model.person;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
  * Represents a module code of a NUS module in the address book.
  * Guarantees: immutable
  */
 public class ModuleCode {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Module code should only contain alphanumeric characters, and it should not be blank.\n"
+            + "The accepted pattern is [Alphabets][Numbers][Alphabets], where the ending [Alphabets] are optional.\n"
+                    + "Note that input check is case-insensitive and program will automatically convert valid inputs\n"
+                    + "to upper case letters for consistent format and easy search of module codes."
+                    + "For example: MA1522, cs2106, CS1101S, is1108, CS2103t are all considered valid inputs.";
+
+    /*
+     * The input should consist of only alphanumeric characters without spaces, should not contain
+     * space, and should not be blank.
+     */
+    public static final String VALIDATION_REGEX = "[A-Za-z]+[0-9]+[A-Za-z]?";
```

Comment:
I realize the suffix length may be more than 1 for a lot of courses (472 courses in this AY). Among them, there are courses that may be taken by Y1 CS students, for example 'CFG2002ID', 'CFG2002KH', 'CFG2002PH', 'CFG2002TH' and 'CFG2002VN' of the Global Industry Insights series where the suffix represents the destination country. Maybe we can change `?` to `{0, 2}` to allow such module codes.
```suggestion
    public static final String VALIDATION_REGEX = "[A-Za-z]+[0-9]+[A-Za-z]{0, 2}";
```

Reply:
Hi, thanks for the review! I confused myself with the meaning of `?` in regex and didn't write test case to cover this kind of edge cases. Thank you so much for spotting that! However, instead of `{0, 2}`, would it be better to replace it with `*`(match 0 or more) instead? Because in some rare cases there may still be suffix with more letters and I think it make sense to handle that kinds of inputs as well. Thank you!